app/routines/generate_reset_code.rb

```diff
@@ -0,0 +1,18 @@
+class GenerateResetCode
```

Comment:
Renaming it is fine.

But in this case the Identity class would be calling the reset method in `PasswordResetCode`, so it has to know about the methods in `PasswordResetCode`. How is this much better than calling `destroy`, which is a standard method for all ActiveRecord? :)